src/DataAPI.jl

```diff
@@ -169,6 +169,34 @@ struct Cols{T<:Tuple}
     Cols(args...) = new{typeof(args)}(args)
 end
 
+"""
+    BroadcastedSelector(selector)
+
+Wrapper type around a `Between`, `All` or `Cols` indicating that
+an operation should be applied to each column included by the wrapped selector.
+
+# Examples
+```jldoctest
+julia> using DataAPI
+
+julia> DataAPI.Between(:a, :e) .=> sin
+DataAPI.BroadcastedSelector{DataAPI.Between{Symbol, Symbol}}(DataAPI.Between{Symbol, Symbol}(:a, :e)) => sin
+
+julia> DataAPI.Cols(r"x") .=> [sum, prod]
```

Comment:
```suggestion
julia> DataAPI.Cols(r"x") .=> [sum prod]
```
using `[sum, prod]` is probably not what the user will want (also the output needs to be changed)

Reply:
ah - I have just checked that this does not look nice in the output, so maybe leave it as is.

Reply:
Contrary to e.g. `["a", "b"] .=> [sin, prod]`, `DataAPI.Cols(r"x") .=> [sum prod]` does apply `sum` and `prod` to all matching columns, right?

Reply:
Yes, the difference is that `["a", "b"] .=> [sin, prod]` broadcasts the same dimension (and we get a vector) while `["a", "b"] .=> [sin prod]` different dimensions (and we get a matrix).

`DataAPI.Cols(r"x")` will be treated as expanding to a vector.